packages/purpose-process/test/clonePurpose.test.ts

```diff
@@ -401,4 +625,222 @@ describe("clonePurpose", async () => {
+      id: purposeCreatedByDelegate.delegationId,
+      kind: delegationKind.delegatedConsumer,
+      eserviceId: purposeCreatedByDelegate.eserviceId,
+      delegatorId: purposeCreatedByDelegate.consumerId,
+      delegateId: consumerDelegate.id,
+      state: delegationState.active,
+    });
+
+    await addOnePurpose(purposeCreatedByDelegate);
+    await addOneTenant(consumer);
+    await addOneTenant(consumerDelegate);
+    await addOneAgreement(mockAgreement);
+    await addOneDelegation(delegation);
+
+    expect(
+      purposeService.clonePurpose({
+        purposeId: purposeCreatedByDelegate.id,
+        seed: {
+          eserviceId: mockEService.id,
+        },
+        ctx: {
+          authData: getRandomAuthData(consumer.id),
+          correlationId: generateId(),
+          logger: genericLogger,
+          serviceName: "test",
+        },
+      })
+    ).rejects.toThrowError(
+      organizationIsNotTheDelegatedConsumer(consumer.id, delegation.id)
+    );
+  });
+  it("should throw organizationIsNotTheConsumer when the requester is the Consumer with no delegation and is cloning a purpose created by a delegate in clonePurpose", async () => {
+    const consumer = {
+      ...getMockTenant(),
+      kind: tenantKind.PA,
+    };
+
+    const consumerDelegate = {
+      ...getMockTenant(),
+      kind: tenantKind.PA,
+    };
+
+    const mockEService = getMockEService();
+
+    const mockAgreement = getMockAgreement(
+      mockEService.id,
+      consumer.id,
+      agreementState.active
+    );
+
+    const purposeCreatedByDelegate: Purpose = {
+      ...getMockPurpose(),
+      eserviceId: mockEService.id,
+      consumerId: consumer.id,
+      versions: [getMockPurposeVersion(purposeVersionState.active)],
+    };
+
+    await addOnePurpose(purposeCreatedByDelegate);
+    await addOneTenant(consumer);
+    await addOneTenant(consumerDelegate);
+    await addOneAgreement(mockAgreement);
+
+    expect(
+      purposeService.clonePurpose({
+        purposeId: purposeCreatedByDelegate.id,
+        seed: {
+          eserviceId: mockEService.id,
+        },
+        ctx: {
+          authData: getRandomAuthData(consumerDelegate.id),
+          correlationId: generateId(),
+          logger: genericLogger,
+          serviceName: "test",
+        },
+      })
+    ).rejects.toThrowError(organizationIsNotTheConsumer(consumerDelegate.id));
+  });
+  it("should throw organizationIsNotTheConsumer if the requester is a delegate for the eservice and there is no delegationId in the purpose", async () => {
+    const consumer = {
+      ...getMockTenant(),
+      kind: tenantKind.PA,
+    };
+
+    const consumerDelegate = {
+      ...getMockTenant(),
+      kind: tenantKind.PA,
+    };
+
+    const mockEService = getMockEService();
+
+    const mockAgreement = getMockAgreement(
+      mockEService.id,
+      consumer.id,
+      agreementState.active
+    );
+
+    const purposeCreatedByDelegate: Purpose = {
```

Comment:
Rename, this is not create by the delegate (delegationId: undefined)
```suggestion
    const purpose: Purpose = {
```